src/components/Explore/TracesByService/Tabs/TabsBarScene.tsx

```diff
@@ -51,6 +52,19 @@ export class TabsBarScene extends SceneObjectBase<TabsBarSceneState> {
       enabledViews = actionViewsDefinitions;
     }
 
+    useEffect(() => {
+      if (hasSetView.current) {
+        return;
+      }
+
+      // Set the view to traceList if the data is loaded and the traces count is greater than 20
+      if (exploration.state.embedded && dataState.data?.state === LoadingState.Done && tracesCount !== undefined && tracesCount > 20) {
+        hasSetView.current = true;
+        metricScene.setActionView('traceList');
+        return;
+      }
+    }, [dataState.data?.state, tracesCount]);
```

Comment:
I haven't tested it, but could it inadvertently switch to traceList from some other view when the data is refreshed?